src/main/scala/schemas.scala

```diff
@@ -5,6 +5,13 @@ package schema
 object ScalaSchema {
   type Prim[A] = ???
 }
+
 object JsonSchema {
-  type Prim[A] = ???
+  type Prim[A] = JsonPrim[A]
```

Comment:
The `A` type parameter here denotes the type that is represented by a given schema (see the binding [here](https://github.com/scalaz/scalaz-schema/blob/53c191b47e1f115c45813a467620b9fe0eac36c8/src/main/scala/SchemaModule.scala#L18) )
So IMO you don't need it in the definition of `JsonPrim`.

Reply:
BTW @jdegoes, can you think of a case where the type parameter of `Prim[A]` is needed.

From what I understand, within the same module implementation, the set of primitive types will be the same for all schema values.

Reply:
Well if you have `SchemaPrim(PrimInt: Prim[Int])`, you'd like to be able to get `Gen[Int]`.